model/src/util.py

```diff
@@ -99,8 +113,14 @@ def __init__(self, tau_mean: float = 600) -> None:
         """
         self.mean: Optional[torch.Tensor] = None
         self.tau_mean = tau_mean
+        self.spike_rec: Deque[torch.Tensor] = deque(maxlen=MAX_RETAINED_SPIKES)
```

Comment:
Since the `SpikeMovingAverage` abstraction was already processing spikes, I have this deque in here to store the last two spikes. 

This is good because it doesn't need to be tracked outside, however causes a bit of weirdness because the mem still needs to be tracked. That has been left in `Layer`. So now they are in two places, but I think it is fine.